libraries/SITL/SIM_Aircraft.cpp

```diff
@@ -553,37 +547,37 @@ bool Aircraft::get_mag_field_ef(float latitude_deg, float longitude_deg, float &
     bool valid_input_data = true;
 
     /* round down to nearest sampling resolution */
-    int min_lat = (int)(latitude_deg / SAMPLING_RES) * SAMPLING_RES;
-    int min_lon = (int)(longitude_deg / SAMPLING_RES) * SAMPLING_RES;
+    int32_t min_lat = static_cast<int32_t>(static_cast<int32_t>(latitude_deg / SAMPLING_RES) * SAMPLING_RES);
+    int32_t min_lon = static_cast<int32_t>(static_cast<int32_t>(longitude_deg / SAMPLING_RES) * SAMPLING_RES);
```

Comment:
Why two casts? Why not allow it to do all operations in float and only in the end cast it to int32?